projects/portal/src/app/pages/vote/proposals/proposal/proposal.usecase.service.ts

```diff
@@ -0,0 +1,66 @@
+import { txParseProposalContent } from './../../../../../../../explorer/src/app/utils/tx-parser';
+import { CosmosRestService } from './../../../../models/cosmos-rest.service';
+import { Injectable } from '@angular/core';
+import cosmosclient from '@cosmos-client/core';
+import {
+  InlineResponse20026DepositParams,
+  InlineResponse20026TallyParams,
+  InlineResponse20026VotingParams,
+  InlineResponse20027FinalTallyResult,
+  InlineResponse20027Proposals,
+  InlineResponse20029Deposits,
+  InlineResponse20032Votes,
+} from '@cosmos-client/core/esm/openapi';
+import { Observable } from 'rxjs';
+import { map, mergeMap } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ProposalUseCaseService {
+  constructor(private readonly cosmosRest: CosmosRestService) {}
+
+  proposal$(proposalID$: Observable<string>): Observable<InlineResponse20027Proposals | undefined> {
+    //console.log('cosmosRest', this.cosmosRest); //----> for debug
+    return proposalID$.pipe(mergeMap((id) => this.cosmosRest.getProposal$(id)));
+  }
+  proposalType$(
+    proposal$: Observable<InlineResponse20027Proposals | undefined>,
+  ): Observable<string | undefined> {
+    return proposal$.pipe(
+      map((proposal) => {
+        //console.log('prop_debug', proposal);
+        if (proposal && proposal.content) {
+          return (proposal.content as any)['@type'];
```

Comment:
@yutaro-elk 
How do I make a mock of the object that have '@type' property?

Reply:
It's okay if you surround it with "

```ts
      const mockContent: InlineResponse200Accounts = {
        "@type": ""
        type_url: 'test_url',
        value: 'test_value',
      };
```